File: bluetooth/wifi/wifi-service.js
var util = require('util');
var bleno = require('bleno');

var WifiState = require('./wifi-state');

var WifiSSIDCharacteristic = require('./wifi-ssid-characteristic');
var WifiPSKCharacteristic = require('./wifi-psk-characteristic');
var WifiUpdateCharacteristic = require('./wifi-update-characteristic');

function WifiService() {
    var wifiState = new WifiState();
    bleno.PrimaryService.call(this, {
        uuid: '08c7042c-12da-49e8-845e-6086d18a81fa',
        characteristics: [
            new WifiSSIDCharacteristic(wifiState),
            new WifiPSKCharacteristic(wifiState),
            new WifiUpdateCharacteristic(wifiState),
        ],
    });
}

util.inherits(WifiService, bleno.PrimaryService);

module.exports = WifiService;

File: bluetooth/wifi/wifi_service.py
from pybleno import *
from .wifi_ssid_characteristic import SSIDCharacteristic
from .wifi_psk_characteristic import PSKCharacteristic
from .wifi_update_characteristic import WiFiUpdateCharacteristic
from .wifi_feedback_characteristic import WiFiFeedbackCharacteristic


class WifiService(BlenoPrimaryService):
    uuid = "3001A7D3-D8A4-4FEA-8174-1736E808C066"

    def __init__(self, wifi_state):
        self._wifi_state = wifi_state
        BlenoPrimaryService.__init__(self, {
            'uuid': self.uuid,
            'characteristics': [
                SSIDCharacteristic(wifi_state),
                PSKCharacteristic(wifi_state),
                WiFiUpdateCharacteristic(wifi_state),
                WiFiFeedbackCharacteristic(wifi_state),
            ]
        })
        print("Started WiFi Service")
